### PR TITLE
Refresh token lookup based off entity instead of the key to account for authority aliases

### PR DESCRIPTION
**Pull request by sameerag**
`RefreshToken` lookup today is based off generating a RefreshToken key and lookup based off the Key. Since we are sharing cache with `python` and other libraries which may not always respect the `preferredCacheAuthority` to cache the authority key (`environment`) look up based off the RT key may fail in some cases.

This PR modifies the lookup to look into the entity which accounts for the aliases for the `environment`

### CODE REVIEW

File: lib/msal-common/src/cache/CacheManager.ts [hunk 1]
```diff
@@ -239,18 +239,28 @@ export abstract class CacheManager implements ICacheManager {
...
+        const refreshTokenFilter: CredentialFilter = {
+            homeAccountId: account.homeAccountId,
+            environment: account.environment,
+            credentialType: CredentialType.REFRESH_TOKEN,
             clientId
-        );
+        };
+
+        const credentialCache: CredentialCache = this.getCredentialsFilteredBy(refreshTokenFilter);
+        const refreshTokens = Object.keys(credentialCache.accessTokens).map((key) => credentialCache.refreshTokens[key]);
```
Comment (tnorling) [inline lib/msal-common/src/cache/CacheManager.ts:254]: This should be refreshTokens right?
```suggestion
        const refreshTokens = Object.keys(credentialCache.refreshTokens).map((key) => credentialCache.refreshTokens[key]);
```

File: lib/msal-common/src/cache/CacheManager.ts [hunk 2]
```diff
@@ -239,18 +239,28 @@ export abstract class CacheManager implements ICacheManager {
...
+
+        const numAccessTokens = refreshTokens.length;
```
Comment (tnorling) [inline lib/msal-common/src/cache/CacheManager.ts:256]:
```suggestion
        const numRefreshTokens = refreshTokens.length;
```

File: lib/msal-common/src/cache/CacheManager.ts (hunk 1 above)
Comment (sameerag) [reply lib/msal-common/src/cache/CacheManager.ts:254]: Yes, sorry missed a checkin. Done

File: lib/msal-common/src/cache/CacheManager.ts (hunk 2 above)
Comment (sameerag) [reply lib/msal-common/src/cache/CacheManager.ts:256]: Yes, sorry missed a checkin. Done

**Review by sangonzal — APPROVED** [review]
Looks good, please add tests ensuring changes meet requirements.

File: lib/msal-common/src/cache/CacheManager.ts [hunk 3]
```diff
@@ -239,18 +239,27 @@ export abstract class CacheManager implements ICacheManager {
...
+        };
+
+        const credentialCache: CredentialCache = this.getCredentialsFilteredBy(refreshTokenFilter);
+        const refreshTokens = Object.keys(credentialCache.refreshTokens).map((key) => credentialCache.refreshTokens[key]);
+
+        const numrefreshTokens = refreshTokens.length;
+        if (numrefreshTokens < 1) {
+            return null;
+        }
+        // Handle this case later - after modifying removeAccount based on environment aliases
```
Comment (tnorling) [inline lib/msal-common/src/cache/CacheManager.ts:260]: Should multiple matches throw an error?

File: lib/msal-common/test/cache/CacheManager.spec.ts [hunk 4]
```diff
@@ -464,4 +464,20 @@ describe("CacheManager.ts test cases", () => {
...
+        ClientTestUtils.setCloudDiscoveryMetadataStubs();
+        const mockedAccountInfo: AccountInfo = {
+            homeAccountId: "uid.utid",
+            environment: "login.windows.net",
+            tenantId: "mocked_tid",
+            username: "mocked_username"
+        };
+
+        const cachedToken = cacheManager.getRefreshTokenEntity("mock_client_id", mockedAccountInfo);
+        console.log(cachedToken);
```
Comment (tnorling) [inline lib/msal-common/test/cache/CacheManager.spec.ts:478]: console.log

**Review by tnorling — APPROVED** [review]
Question and a nit but looks good otherwise

**Comment by sameerag** [comment]
#2201 addresses this - merged all changes including tests. Closing this PR.

**Pull request closed without merging**